[ExecuTorch] Add broadcasting support to optimized op_div

kimishpatel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #8257

Summary:
Similar to broadcast support in op_mul

Test Plan:
Tests added

Reviewers:

Subscribers:

Tasks:

Tags:

cc @larryliu0820 @manuelcandales

Differential Revision: [D69491815](https://our.internmc.facebook.com/intern/diff/D69491815)

kimishpatel committed:
[ExecuTorch] Add broadcasting support to optimized op_div

Summary:
Similar to broadcast support in op_mul

Test Plan:
Tests added

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

kimishpatel committed:
Update on "[ExecuTorch] Add broadcasting support to optimized op_div"


Summary:
Similar to broadcast support in op_mul

Test Plan:
Tests added

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

kimishpatel committed:
Update on "[ExecuTorch] Add broadcasting support to optimized op_div"


Summary:
Similar to broadcast support in op_mul

Test Plan:
Tests added

Reviewers:

Subscribers:

Tasks:

Tags:

cc larryliu0820 manuelcandales

[ghstack-poisoned]

kimishpatel committed:
Update on "[ExecuTorch] Add broadcasting support to optimized op_div"


Summary:
Similar to broadcast support in op_mul

Test Plan:
Tests added

Reviewers:

Subscribers:

Tasks:

Tags:

cc larryliu0820 manuelcandales

[ghstack-poisoned]

kimishpatel committed:
Update on "[ExecuTorch] Add broadcasting support to optimized op_div"


Summary:
Similar to broadcast support in op_mul

Test Plan:
Tests added

Reviewers:

Subscribers:

Tasks:

Tags:

cc larryliu0820 manuelcandales

[ghstack-poisoned]

kimishpatel committed:
Update on "[ExecuTorch] Add broadcasting support to optimized op_div"


Summary:
Similar to broadcast support in op_mul

Test Plan:
Tests added

Reviewers:

Subscribers:

Tasks:

Tags:

cc larryliu0820 manuelcandales

[ghstack-poisoned]

swolchok commented on kernels/optimized/cpu/op_div.cpp:
```diff
@@ -120,46 +120,23 @@ Tensor& opt_div_out(
+    // Reason for using alpha is becasuse handle_broadcast_elementwise
+    // is used for add and sub as well:
+    static constexpr const char op_name[] = "div.out";
+    ET_SWITCH_REALB_TYPES(out_type, ctx, "mul.out", CTYPE, [&]() {
```
s/mul/div/ here

kimishpatel replied:
good catch

swolchok reviewed (approved):
make sure to fix the use of mul.out. op_name is also unused

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D69491815).

kimishpatel committed:
Update on "[ExecuTorch] Add broadcasting support to optimized op_div"


Summary:
Similar to broadcast support in op_mul

Test Plan:
Tests added

Reviewers:

Subscribers:

Tasks:

Tags:

cc larryliu0820 manuelcandales

Differential Revision: [D69491815](https://our.internmc.facebook.com/intern/diff/D69491815)

[ghstack-poisoned]

kimishpatel committed:
Update on "[ExecuTorch] Add broadcasting support to optimized op_div"


Summary:
Similar to broadcast support in op_mul

Test Plan:
Tests added

Reviewers:

Subscribers:

Tasks:

Tags:

cc larryliu0820 manuelcandales

Differential Revision: [D69491815](https://our.internmc.facebook.com/intern/diff/D69491815)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D69491815).

kimishpatel committed:
Update on "[ExecuTorch] Add broadcasting support to optimized op_div"


Summary:
Similar to broadcast support in op_mul

Test Plan:
Tests added

Reviewers:

Subscribers:

Tasks:

Tags:

cc larryliu0820 manuelcandales

Differential Revision: [D69491815](https://our.internmc.facebook.com/intern/diff/D69491815)

[ghstack-poisoned]

kimishpatel committed:
Update on "[ExecuTorch] Add broadcasting support to optimized op_div"


Summary:
Similar to broadcast support in op_mul

Test Plan:
Tests added

Reviewers:

Subscribers:

Tasks:

Tags:

cc larryliu0820 manuelcandales

Differential Revision: [D69491815](https://our.internmc.facebook.com/intern/diff/D69491815)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D69491815).

kimishpatel committed:
Update on "[ExecuTorch] Add broadcasting support to optimized op_div"


Summary:
Similar to broadcast support in op_mul

Test Plan:
Tests added

Reviewers:

Subscribers:

Tasks:

Tags:

cc larryliu0820 manuelcandales

Differential Revision: [D69491815](https://our.internmc.facebook.com/intern/diff/D69491815)

[ghstack-poisoned]

kimishpatel committed:
Update on "[ExecuTorch] Add broadcasting support to optimized op_div"


Summary:
Similar to broadcast support in op_mul

Test Plan:
Tests added

Reviewers:

Subscribers:

Tasks:

Tags:

cc larryliu0820 manuelcandales

Differential Revision: [D69491815](https://our.internmc.facebook.com/intern/diff/D69491815)

[ghstack-poisoned]

kimishpatel committed:
Update on "[ExecuTorch] Add broadcasting support to optimized op_div"


Summary:
Similar to broadcast support in op_mul

Test Plan:
Tests added

Reviewers:

Subscribers:

Tasks:

Tags:

cc larryliu0820 manuelcandales

Differential Revision: [D69491815](https://our.internmc.facebook.com/intern/diff/D69491815)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D69491815).

kimishpatel committed:
Update on "[ExecuTorch] Add broadcasting support to optimized op_div"


Summary:
Similar to broadcast support in op_mul

Test Plan:
Tests added

Reviewers:

Subscribers:

Tasks:

Tags:

cc larryliu0820 manuelcandales

Differential Revision: [D69491815](https://our.internmc.facebook.com/intern/diff/D69491815)

[ghstack-poisoned]

kimishpatel committed:
Update on "[ExecuTorch] Add broadcasting support to optimized op_div"


Summary:
Similar to broadcast support in op_mul

Test Plan:
Tests added

Reviewers:

Subscribers:

Tasks:

Tags:

cc larryliu0820 manuelcandales

Differential Revision: [D69491815](https://our.internmc.facebook.com/intern/diff/D69491815)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D69491815).

kimishpatel committed:
Update on "[ExecuTorch] Add broadcasting support to optimized op_div"


Summary:
Similar to broadcast support in op_mul

Test Plan:
Tests added

Reviewers:

Subscribers:

Tasks:

Tags:

cc larryliu0820 manuelcandales

Differential Revision: [D69491815](https://our.internmc.facebook.com/intern/diff/D69491815)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D69491815).

kimishpatel committed:
Update on "[ExecuTorch] Add broadcasting support to optimized op_div"


Summary:
Similar to broadcast support in op_mul

Test Plan:
Tests added

Reviewers:

Subscribers:

Tasks:

Tags:

cc larryliu0820 manuelcandales

Differential Revision: [D69491815](https://our.internmc.facebook.com/intern/diff/D69491815)

[ghstack-poisoned]

kimishpatel commented:
@kimishpatel has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D69491815).
